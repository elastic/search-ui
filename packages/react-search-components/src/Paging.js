import PropTypes from "prop-types";
import React from "react";
import RCPagination from "rc-pagination";

function Paging({ current, resultsPerPage, onChange, totalPages }) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className="sui-paging"
    />
  );
}

Paging.propTypes = {
  current: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired
};

export default Paging;
