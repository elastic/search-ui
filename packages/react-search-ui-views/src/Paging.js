import PropTypes from "prop-types";
import React from "react";
import RCPagination from "rc-pagination";
import enUsLocale from "rc-pagination/lib/locale/en_US";

import { appendClassName } from "./view-helpers";

function Paging({
  className,
  current,
  resultsPerPage,
  onChange,
  totalPages,
  ...rest
}) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className={appendClassName("sui-paging", className)}
      locale={enUsLocale}
      {...rest}
    />
  );
}

Paging.propTypes = {
  current: PropTypes.number.isRequired,
  onChange: PropTypes.func.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  className: PropTypes.string
};

export default Paging;
