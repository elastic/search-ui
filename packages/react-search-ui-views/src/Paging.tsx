import React from "react";
import RCPagination from "rc-pagination";
import enUsLocale from "rc-pagination/lib/locale/en_US";

import { appendClassName } from "./view-helpers";
import type { SearchContextState } from "@elastic/search-ui";
import { Rename, BaseContainerProps } from "./types";

export type PagingContainerContext = Pick<
  SearchContextState,
  "current" | "resultsPerPage" | "totalPages" | "setCurrent"
>;

export type PagingViewProps = Rename<
  BaseContainerProps & PagingContainerContext,
  {
    setCurrent: "onChange";
  }
>;

export type PagingContainerProps = BaseContainerProps &
  PagingContainerContext & {
    view?: React.ComponentType<PagingViewProps>;
  };

function Paging({
  className,
  current,
  resultsPerPage,
  onChange,
  totalPages,
  ...rest
}: PagingViewProps) {
  return (
    <RCPagination
      current={current}
      onChange={onChange}
      pageSize={resultsPerPage}
      total={totalPages * resultsPerPage}
      className={appendClassName("sui-paging", className)}
      locale={enUsLocale}
      {...rest}
    />
  );
}

export default Paging;
