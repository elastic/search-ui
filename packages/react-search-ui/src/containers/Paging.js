import PropTypes from "prop-types";

import { withSearch } from "..";
import { Paging } from "@elastic/react-search-ui-views";

export function PagingContainer({
  className,
  current,
  resultsPerPage,
  setCurrent,
  totalPages,
  view,
  ...rest
}) {
  if (totalPages === 0) return null;

  const View = view || Paging;

  return View({
    className,
    current,
    resultsPerPage,
    totalPages,
    onChange: setCurrent,
    ...rest
  });
}

PagingContainer.propTypes = {
  // Props
  className: PropTypes.string,
  view: PropTypes.func,
  // State
  current: PropTypes.number.isRequired,
  resultsPerPage: PropTypes.number.isRequired,
  totalPages: PropTypes.number.isRequired,
  // Action
  setCurrent: PropTypes.func.isRequired
};

export default withSearch(
  ({ current, resultsPerPage, totalPages, setCurrent }) => ({
    current,
    resultsPerPage,
    totalPages,
    setCurrent
  })
)(PagingContainer);
