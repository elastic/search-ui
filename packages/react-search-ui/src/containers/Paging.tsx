import React from "react";

import { withSearch } from "..";
import {
  Paging,
  PagingContainerContext,
  PagingContainerProps,
  PagingViewProps
} from "@elastic/react-search-ui-views";

export function PagingContainer({
  className,
  current,
  resultsPerPage,
  setCurrent,
  totalPages,
  view,
  ...rest
}: PagingContainerProps) {
  if (totalPages === 0) return null;

  const View: React.ComponentType<PagingViewProps> = view || Paging;

  const viewProps = {
    className,
    current,
    resultsPerPage,
    totalPages,
    onChange: setCurrent,
    ...rest
  };

  return <View {...viewProps} />;
}

export default withSearch<PagingContainerProps, PagingContainerContext>(
  ({ current, resultsPerPage, totalPages, setCurrent }) => ({
    current,
    resultsPerPage,
    totalPages,
    setCurrent
  })
)(PagingContainer);
