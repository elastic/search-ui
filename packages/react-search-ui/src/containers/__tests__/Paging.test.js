import React from "react";
import { shallow } from "enzyme";
import { PagingContainer } from "../Paging";

const params = {
  current: 1,
  resultsPerPage: 20,
  setCurrent: jest.fn(),
  totalPages: 5
};

beforeEach(() => {
  params.setCurrent = jest.fn();
});

it("supports a render prop", () => {
  // eslint-disable-next-line react/prop-types
  const render = ({ current }) => {
    return <div>{current}</div>;
  };
  const wrapper = shallow(<PagingContainer {...params} view={render} />);
  expect(wrapper).toMatchSnapshot();
});

it("renders empty when there are no results", () => {
  const wrapper = shallow(
    <PagingContainer
      {...{
        ...params,
        totalPages: 0
      }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it("will call back when a the page is changed", () => {
  let viewProps;

  shallow(<PagingContainer {...params} view={props => (viewProps = props)} />);

  const { onChange } = viewProps;
  onChange(2);

  const current = params.setCurrent.mock.calls[0][0];
  expect(current).toEqual(2);
});
