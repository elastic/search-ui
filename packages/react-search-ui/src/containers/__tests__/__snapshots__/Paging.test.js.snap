// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders empty when there are no results 1`] = `""`;

exports[`supports a render prop 1`] = `
<div>
  1
</div>
`;
